### PR TITLE
Parallelizing Raw Atlas Way-Sectioning and Point/Line Slicing

### PR DESCRIPTION
**Pull request by MikeGost**
A few changes here:

1. Fully parallelizes `Point` way-sectioning logic. This reduces total runtime by ~30-35%.
2. Parallelizes `Point` and `Line` slicing logic, this adds an additional 10-15% improvement.
3. Fixes a way-sectioning edge case and adds a test for it
4. Improved logging and time tracking
5. Adds missing serial version UID for StringList

P.S. We need an `Iterables.parallelStream(...)` method - created an issue :)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java [hunk 1]
```diff
@@ -500,8 +501,8 @@ private void distinguishPointsFromShapePoints(final WaySectionChangeSet changeSe
         final Time time = Time.now();
         logger.info("Started Shape Point Detection for Shard {}", getShardOrAtlasName());
 
-        Iterables.stream(this.rawAtlas.points()).filter(point -> isAtlasPoint(changeSet, point))
-                .forEach(changeSet::recordPoint);
+        StreamSupport.stream(this.rawAtlas.points().spliterator(), true)
```
Comment (mgcuthbert) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java]: Out of genuine curiosity, how much performance gain do you get by using spliterator here?

Comment (MikeGost) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java]: ~3x increase in performance.

**Comment by lucaspcram** [comment]
Looks good to me. The `spliterator` performance increases are very cool.
